subsys/bluetooth/host/iso.c

```diff
@@ -1096,3 +1105,424 @@ int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf)
+		bt_iso_chan_add(bis->conn, bis);
+	}
+
+	return 0;
+}
+
+static int hci_le_create_big(struct bt_le_ext_adv *padv,
+			     struct bt_iso_big_create_param_t *param)
+{
+	struct bt_hci_cp_le_create_big *req;
+	struct bt_hci_evt_le_big_complete *evt;
+	struct bt_iso_big *big = param->big;
+	struct net_buf *buf;
+	struct net_buf *rsp;
+	int err;
+	static struct bt_iso_chan_qos *qos;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_CREATE_BIG, sizeof(*req));
+
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	qos = param->big->bis[0]->qos;
+
+	req = net_buf_add(buf, sizeof(*req));
+	req->big_handle = big->handle;
+	req->adv_handle = padv->handle;
+	req->num_bis = big->num_bis;
+	sys_put_le24(qos->interval, req->sdu_interval);
+	req->max_sdu = sys_cpu_to_le16(qos->sdu);
+	req->max_latency = sys_cpu_to_le16(qos->latency);
+	req->rtn = qos->rtn;
+	req->phy = qos->phy;
+	req->packing = qos->packing;
+	req->framing = qos->framing;
+	req->encryption = param->encryption;
+	if (req->encryption) {
+		memcpy(req->bcode, param->bcode, sizeof(req->bcode));
+	} else {
+		memset(req->bcode, 0, sizeof(req->bcode));
+	}
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_CREATE_BIG, buf, &rsp);
+	if (err) {
+		return err;
+	}
+
+	evt = (struct bt_hci_evt_le_big_complete *)rsp->data;
+	if (evt->status ||
+	    evt->big_handle != big->handle ||
+	    evt->num_bis != big->num_bis) {
+		err = -EIO;
+	} else {
+		for (int i = 0; i < big->num_bis; i++) {
+			big->bis[i]->conn->handle = evt->handle[i];
+			bt_iso_chan_set_state(big->bis[i], BT_ISO_CONNECTED);
+		}
+
+		bigs[big->handle] = big;
+	}
+
+	net_buf_unref(rsp);
+
+	return err;
+
+}
+
+int bt_iso_big_create(struct bt_le_ext_adv *padv,
+		      struct bt_iso_big_create_param_t *param)
+{
+	struct bt_iso_big *big = param->big;
+	int err;
+
+	if (!atomic_test_bit(padv->flags, BT_PER_ADV_PARAMS_SET)) {
+		BT_DBG("PA params not set; invalid adv object");
+		return -EINVAL;
+	}
+
+	if (big->handle) {
+		BT_DBG("BIG handle already initialized %u", big->handle);
+		return -EALREADY;
+	}
+
+	err = big_init_bis(big, true);
+	if (err) {
+		BT_DBG("Could not init BIG %d", err);
+		cleanup_big(big);
+		return err;
+	}
+
+	err = hci_le_create_big(padv, param);
+	if (err) {
+		BT_DBG("Could not create BIG %d", err);
+		cleanup_big(big);
+		return err;
+	}
+
+	return err;
+}
+
+static int hci_le_terminate_big(struct bt_iso_big *big)
+{
+	struct bt_hci_cp_le_terminate_big *req;
+	struct bt_hci_evt_le_big_terminate *evt;
+	struct net_buf *buf;
+	struct net_buf *rsp;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_TERMINATE_BIG, sizeof(*req));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	req = net_buf_add(buf, sizeof(*req));
+	req->big_handle = big->handle;
+	req->reason = BT_HCI_ERR_REMOTE_USER_TERM_CONN;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_TERMINATE_BIG, buf, &rsp);
+	if (err) {
+		return err;
+	}
+
+	evt = (struct bt_hci_evt_le_big_terminate *)rsp->data;
+	if (evt->big_handle != big->handle) {
+		err = -EIO;
+	} else {
+		big_disconnect(big);
```

Comment:
Have to wait for HCI LE BIG Terminate Complete event.

Reply:
Interesting that I didn't implement that. Reimplementing it feels like I did that already, but for some reason must have removed that code again. 

Anyways, I readded the handling of the `BT_HCI_EVT_LE_BIG_TERMINATE` event such that we, as the broadcaster, don't clean up the BIG in the host until we receive that event.